File: Elixir/0002.ex
defmodule FibonacciSum do
  def sum_even_fibonacci(limit) do
    sum_even_fibonacci(1, 2, 0, limit)
  end

  defp sum_even_fibonacci(a, b, sum, limit) when a <= limit do
    new_sum = if rem(a, 2) == 0, do: sum + a, else: sum
    sum_even_fibonacci(b, a + b, new_sum, limit)
  end

  defp sum_even_fibonacci(_a, _b, sum, _limit) do
    sum
  end
end

defmodule Main do
  def run do
    limit = 4000000
    result = FibonacciSum.sum_even_fibonacci(limit)
    IO.puts("The sum of the even-valued terms is: #{result}")
  end
end

Main.run()

File: Erlang/0002.erl
-module(fibonacci_sum).
-export([main/1]).

% Entry point for the escript
main(Args) ->
    Limit = 4000000,
    Sum = sum_even_fibonacci(Limit),
    io:format("The sum of the even-valued terms below ~p is: ~p~n", [Limit, Sum]).

% Function to calculate the sum of even Fibonacci numbers below a given limit
sum_even_fibonacci(Limit) ->
    sum_even_fibonacci(1, 2, 0, Limit).

% Recursive function to accumulate the sum of even Fibonacci numbers
sum_even_fibonacci(A, B, Acc, Limit) when A =< Limit ->
    NewAcc = if 
        A rem 2 =:= 0 -> Acc + A;
        true -> Acc
    end,
    NextA = B,
    NextB = A + B,
    sum_even_fibonacci(NextA, NextB, NewAcc, Limit);
sum_even_fibonacci(_, _, Acc, _) ->
    Acc.
